openstf/model/ml_model/abstract_ml_model.py

```diff
@@ -0,0 +1,84 @@
+# SPDX-FileCopyrightText: 2017-2021 Alliander N.V. <korte.termijn.prognoses@alliander.com> # noqa E501>
+#
+# SPDX-License-Identifier: MPL-2.0
+
+from abc import ABC, abstractmethod
+import pandas as pd
+import structlog
+from typing import Union
+import numpy as np
+from sklearn.base import RegressorMixin, BaseEstimator
+
+
+class AbstractMLModel(BaseEstimator, RegressorMixin, ABC):
+    """
+    This class defines an abstract ML model,
+    of which concrete implementation can be used as a workhorse throughout openstf
+
+    If you want to include a new ML model in openstf, it should adhere to the interface
+    defined here
+    """
+
+    def __init__(self) -> None:
+        self.logger = structlog.get_logger(self.__class__.__name__)
+
+    @abstractmethod
+    def predict(self, x: pd.DataFrame) -> np.array:
+        pass
+
+    @abstractmethod
+    def fit(self, x: np.array, y: np.array, **kwargs) -> RegressorMixin:
+        pass
+
+    def score(self, forecast: np.array, actual: np.array) -> float:
```

Comment:
overweeg of de implementatie ook in de abtracte klasse hoort. 
Alternatief: hier echt alleen de interface en dit soort implementatie in een tussenliggende base-klasse o.i.d.